conmon/conmon.c

```diff
@@ -618,7 +620,23 @@ static bool read_stdio(int fd, stdpipe_t pipe, bool *eof)
 
 static void on_sigchld(G_GNUC_UNUSED int signal)
 {
-	raise (SIGUSR1);
+	raise(SIGUSR1);
+}
+
+static void on_sig_exit(int signal)
+{
+	if (container_pid > 0) {
+		if (kill(container_pid, signal) == 0)
+			return;
+	} else if (create_pid > 0) {
+		if (kill(create_pid, signal) < 0 && errno == ESRCH) {
+			/* The create_pid process might have exited, so try container_pid again.  */
+			if (container_pid > 0 && kill(container_pid, signal) == 0)
+				return;
+		}
+	}
+	/* We should never get here, just force a check if it happens.   */
```

Comment:
You can get here if `container_pid` is not yet set and `kill(create_pid, signal)` succeeds, or if that kill errors other than `ESRCH`.  In those cases I think you want to qietly exit the handler instead of raising a signal.

I also think we should reset the PUD globals to `-1` in ghe respectice exit callbacks.  Otherwise we'll attempt to signal the container process if we recieve a `SIGTERM` after reaping the container process.